include/stacktracetools.h

```diff
@@ -0,0 +1,6 @@
+#if MIN_VERSION_base(4,9,0)
+import GHC.Stack (HasCallStack)
+#define CHECK(f) (withFrozenCallStack Ck.f __FILE__ __LINE__)
+#else
+#define HasCallStack (Eq ())
```

Comment:
That `Eq ()` fallback constraint was just a rough hack, to replace `HasCallStack` with something as simple as possible on GHC versions that don't support it yet. It turns out that hack doesn't work:
```
Data/Vector/Internal/Check.hs:122:15:
    Non type-variable argument in the constraint: Eq ()
    (Use FlexibleContexts to permit this)
    In the type signature for ‘checkIndex’:
      checkIndex :: (Eq ()) =>
                    String -> Int -> Checks -> String -> Int -> Int -> a -> a
```
Any ideas how this could be properly done without cluttering every signature with an `#if MIN_VERSION_base`?

Reply:
I would do what the [`call-stack` library does](http://hackage.haskell.org/package/call-stack-0.1.0/docs/src/Data-CallStack.html):

```haskell
#if MIN_VERSION_base(4,9,0)
import           GHC.Stack (HasCallStack)
#elif MIN_VERSION_base(4,8,1)
type HasCallStack = (?callStack :: GHC.CallStack)
#else
import GHC.Exts (Constraint)
type HasCallStack = (() :: Constraint)
#endif
```

This requires `base-4.5` (GHC 7.4) or later, but those are precisely the version bounds that `vector` has. Moreover, this shouldn't require the use of `FlexibleContexts`.

Reply:
I don't suppose it would be ok for `vector` to actually depend on `call-stack`?